sqlmesh/core/engine_adapter/logical_merge.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp, parse_one
+
+from sqlmesh.core.engine_adapter.base import EngineAdapter, TransactionType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import QueryOrDF
+
+
+class LogicalMerge(EngineAdapter):
+    def merge(
+        self,
+        target_table: TableName,
+        source_table: QueryOrDF,
+        columns_to_types: t.Dict[str, exp.DataType],
+        unique_key: t.Sequence[str],
+    ) -> None:
+        """
+        Merge implementation for engine adapters that do not support merge natively.
+
+        The merge is executed as follows:
+        1. Create a temporary table containing the new data to merge.
+        2. Delete rows from target table where unique_key cols match a row in the temporary table.
+        3. Insert the temporary table contents into the target table. Any duplicate, non-unique rows
+           within the temporary table are ommitted.
+        4. Drop the temporary table.
+        """
+        temp_table = self._get_temp_table(target_table)
+        unique_exp: exp.Expression = parse_one("|| '__SQLMESH_DELIM__' ||".join(unique_key))
```

Comment:
can you do instead

exp.func("CONCAT_WS", "__SQLMESH_DELIM__", *unique_key)

Reply:
Yeah, good idea.